File: backend/src/models/attendance.rs
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Debug, Deserialize, Serialize, sqlx::FromRow)]
pub struct Attendance {
    pub id: Uuid,
    pub person_id: Uuid,
    pub date: chrono::NaiveDate,
    pub status: String, // present, absent, late, excused
    pub time: Option<chrono::NaiveTime>,
    pub created_at: chrono::NaiveDateTime,
    pub updated_at: chrono::NaiveDateTime,
}

#[derive(Debug, Deserialize, Serialize)]
pub struct AttendanceCreate {
    pub person_id: Uuid,
    pub date: chrono::NaiveDate,
    pub status: String,
    pub time: Option<chrono::NaiveTime>,
}

#[derive(Debug, Deserialize, Serialize)]
pub struct AttendanceUpdate {
    pub status: Option<String>,
    pub time: Option<chrono::NaiveTime>,
}

#[derive(Debug, Deserialize, Serialize)]
pub struct AttendanceResponse {
    pub id: Uuid,
    pub person_id: Uuid,
    pub date: chrono::NaiveDate,
    pub status: String,
    pub time: Option<chrono::NaiveTime>,
}

impl From<Attendance> for AttendanceResponse {
    fn from(attendance: Attendance) -> Self {
        Self {
            id: attendance.id,
            person_id: attendance.person_id,
            date: attendance.date,
            status: attendance.status,
            time: attendance.time,
        }
    }
}

File: frontend/src/api/attendance.ts
import api from './index'

// 考勤查询参数
export interface AttendanceQuery {
  page: number
  limit: number
  name?: string
  date?: string
  status?: string
}

// 考勤创建参数
export interface AttendanceCreate {
  person_id: string
  date: string
  status: 'present' | 'late' | 'absent' | 'early_leave' | 'excused'
  time?: string
  remark?: string
}

// 考勤更新参数
export interface AttendanceUpdate {
  status?: 'present' | 'late' | 'absent' | 'early_leave' | 'excused'
  time?: string
  remark?: string
}

// 考勤响应数据
export interface AttendanceResponse {
  id: string
  person_id: string
  person_name: string
  date: string
  status: 'present' | 'late' | 'absent' | 'early_leave' | 'excused'
  time?: string
  remark?: string
  created_at: string
}

// 列表响应
export interface ListResponse<T> {
  items: T[]
  total: number
}

// 考勤API
export const attendanceApi = {
  list: (params: AttendanceQuery) => {
    return api.get<ListResponse<AttendanceResponse>>('/attendances', { params })
  },
  create: (data: AttendanceCreate) => {
    return api.post<AttendanceResponse>('/attendances', data)
  },
  get: (id: string) => {
    return api.get<AttendanceResponse>(`/attendances/${id}`)
  },
  update: (id: string, data: AttendanceUpdate) => {
    return api.put<AttendanceResponse>(`/attendances/${id}`, data)
  },
  delete: (id: string) => {
    return api.delete(`/attendances/${id}`)
  }
}
